overlord/hookstate/ctlcmd/model.go

```diff
@@ -0,0 +1,187 @@
+		return mf.GetEscapedDash() + "\033[0m"
+	}
+
+	badge := ""
+	if mf.storeAccount.Validation == "verified" {
+		badge = "*"
+	}
+
+	// NOTE this makes e.g. 'Potato' == 'potato', and 'Potato Team' == 'potato-team',
+	// but 'Potato Team' != 'potatoteam', 'Potato Inc.' != 'potato' (in fact 'Potato Inc.' != 'potato-inc')
+	if strings.EqualFold(strings.Replace(mf.storeAccount.Username, "-", " ", -1), mf.storeAccount.DisplayName) {
+		return mf.storeAccount.DisplayName + badge + "\033[0m"
+	}
+	return fmt.Sprintf("%s (%s%s%s)", mf.storeAccount.DisplayName, mf.storeAccount.Username, badge, "\033[0m")
+}
+
+func newTabWriter(output io.Writer) *tabwriter.Writer {
+	minWidth := 2
+	tabWidth := 2
+	padding := 2
+	padchar := byte(' ')
+	return tabwriter.NewWriter(output, minWidth, tabWidth, padding, padchar, 0)
+}
+
+// reportError prints the error message to stderr
+func (c *modelCommand) reportError(format string, a ...interface{}) {
+	w := newTabWriter(c.stderr)
+	fmt.Fprintf(w, format, a...)
+	w.Flush()
+}
+
+// checkGadgetOrModel verifies that the current snap context is either a gadget snap or that
+// the snap shares the same publisher as the current model assertion.
+func (c *modelCommand) checkGadgetOrModel(st *state.State, snapName string) error {
+	st.Lock()
+	defer st.Unlock()
+
+	var snapst snapstate.SnapState
+	if err := snapstate.Get(st, snapName, &snapst); err != nil {
+		return fmt.Errorf("failed to get snapstate for snap %s: %v", snapName, err)
+	}
+
+	// get the brand of the current snap
+	snapInfo, err := snapst.CurrentInfo()
+	if err != nil {
+		return err
+	}
+
+	deviceCtx, err := devicestate.DeviceCtx(st, nil, nil)
+	if err != nil {
+		return err
+	}
+
+	// the request snap must be a gadget or come from the same
+	// brand as the device model assertion
+	if snapType := snapInfo.Type(); snapType == snap.TypeGadget {
+		return nil
+	}
+	if snapInfo.Publisher.ID == deviceCtx.Model().BrandID() {
+		return nil
+	}
+
+	c.reportError("cannot get model assertion for snap %q: not a gadget or from the same brand as the device model assertion\n", snapName)
+	return fmt.Errorf("insufficient permissions to get model assertion for snap %q", snapName)
+}
+
+func (c *modelCommand) Execute([]string) error {
+
+	context, err := c.ensureContext()
+	if err != nil {
+		return err
+	}
+
+	st := context.State()
+	if err := c.checkGadgetOrModel(st, context.InstanceName()); err != nil {
+		return err
+	}
+
+	st.Lock()
+	deviceCtx, err := snapstate.DeviceCtx(st, nil, nil)
+	if err != nil {
+		return err
+	}
+	st.Unlock()
+
+	// Chosen as the default term width, when none is provided. This seems
+	// like a fair choice to make.
+	termWidth := 80
+	if c.TermWidth > 0 {
+		termWidth = c.TermWidth
+	}
+
+	// use the same tab-writer settings as the 'snap model' in cmd_list.go
+	w := newTabWriter(c.stdout)
+	defer w.Flush()
+
+	opts := clientutil.PrintModelAssertionOptions{
+		TermWidth: termWidth,
+		AbsTime:   false,
+		Verbose:   true,
```

Comment:
I'm not sure if that was already agreed upon - don't we want to have `--abs-time` and `--verbose` flags?

Reply:
I think the spec mentions that verbose is implied, as this for software. I don't know if the spec says something about abs-time, I think we don't want the argument again it should probably be true not false

Reply:
Added comments for this, also fixed the AbsTime, it's now true.